src/components/Steps/Map/index.js

```diff
@@ -55,11 +60,46 @@ class MapStep extends Component {
         <Station lat={station.coordinates.lat} lng={station.coordinates.lon} />
     ));
 
+    displaySearchLinks = areas => areas.map((area, i) => {
+        const { name } = this.props.city;
+        const data = area.map(street =>
+            (<a
+                href={`https://www.olx.pl/nieruchomosci/mieszkania/${CITIES[name]}/q-${street.split(' ').join('-')}`}
+                style={{ textDecoration: 'none' }}
+                target="_blank"
+            >
+                https://www.olx.pl/nieruchomosci/mieszkania/{CITIES[name]}/q-{street.split(' ').join('-')}
+            </a>));
+        return (<Slidedown
+            title={`Area  ${i}`}
+            data={data}
+            closed={this.state.closed[i]}
+            onClick={() => this.handleSlideDown(i)}
+        />);
+    });
 
-    displaySearchLinks = areas => areas.map(area => area.map(street => <span>https://www.olx.pl/nieruchomosci/mieszkania/wroclaw/q-{street.split(' ').join('-')}</span>));
+    displayAreas = stations => stations.map((station, index) =>
+        (<Area
+            lat={station.coordinates.lat}
+            lng={station.coordinates.lon}
+            importance={index}
+            onClick={() => this.handleSlideDown(index)}
+        />));
+
+    handleSlideDown = (id) => {
+        const closed = this.state.closed.slice(0);
```

Comment:
Again, use active... code will be much cleaner and easier to understand. this is magic for me :P